mixed-content/generic/common.js

```diff
@@ -0,0 +1,238 @@
+function normalizePort(targetPort) {
+  var defaultPorts = [80, 443];
+  var isDefaultPortForProtocol = (defaultPorts.indexOf(targetPort) >= 0);
+
+  return (targetPort == "" || isDefaultPortForProtocol) ?
+          "" : ":" + targetPort;
+}
+
+function setAttributes(el, attrs) {
+  attrs = attrs || {}
+  for (var attr in attrs)
+    el[attr] = attrs[attr];
+}
+
+function createElement(tagName, attrs, parent) {
+  var el = document.createElement(tagName);
+  setAttributes(el, attrs);
+  if (parent && parent.appendChild)
+    parent.appendChild(el);
+
+  return el;
+}
+
+function createHelperIframe(name) {
+  return createElement("iframe", {"name": name}, document.body);
+}
+
+function queryIframe(url) {
```

Comment:
Nevermind. It wasn't clear to me that `query*` is what you're using to denote tests. Perhaps `testX` instead?